saleor/dashboard/collection/forms.py

```diff
@@ -45,3 +45,12 @@ class AssignHomepageCollectionForm(forms.ModelForm):
     class Meta:
         model = SiteSettings
         fields = ('homepage_collection',)
+
+    def clean(self):
```

Comment:
I think this should be placed in `clean_homepage_collection` function, what's more, we could change field's queryset to get only collections that are published.